### ISSUE TITLE
On Windows, clj-kondo can report with unix file separators when single file is linted

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#1438]
**version**
2021.10.19

**platform**
native, Windows only

**problem**
Clj-kondo reports paths with the forward slash on Windows when linting single file when requested lint file also uses the forward slash. 

It otherwise reports the Windows backslash file separator.

This seems like a minor issue, but the inconsistent handling does cause issues (case in point, some api-diff tests I am working on).

**repro**
From a Windows CMD shell in an empty dir, create a dummy file that will report something:
```
mkdir sub1
mkdir sub1\sub2
echo (defn- unused []) > sub1\sub2\dummy.clj
```
Now lint with clj-kondo
```
> clj-kondo --lint sub1/sub2/dummy.clj
sub1/sub2/dummy.clj:1:8: warning: Unused private var user/unused
linting took 15ms, errors: 0, warnings: 1
```
Notice the forward slashes.

Linting otherwise seems to consistently report with backslashes on Windows:
```
> clj-kondo --lint sub1
sub1\sub2\dummy.clj:1:8: warning: Unused private var user/unused
linting took 15ms, errors: 0, warnings: 1

> clj-kondo --lint sub1/sub2
sub1\sub2\dummy.clj:1:8: warning: Unused private var user/unused
linting took 16ms, errors: 0, warnings: 1

> clj-kondo --lint sub1\sub2
sub1\sub2\dummy.clj:1:8: warning: Unused private var user/unused
linting took 16ms, errors: 0, warnings: 1

> clj-kondo --lint sub1\sub2\dummy.clj
sub1\sub2\dummy.clj:1:8: warning: Unused private var user/unused
linting took 0ms, errors: 0, warnings: 1
```

The behaviour is the same for analysis data:
```
> clj-kondo --lint sub1/sub2/dummy.clj --config "{:output {:format :edn :analysis true}}" | jet --query :analysis --pretty
{:namespace-definitions [],
 :namespace-usages [],
 :var-definitions [{:fixed-arities #{0},
                    :end-row 1,
                    :name-end-col 14,
                    :name-end-row 1,
                    :private true,
                    :name-row 1,
                    :ns user,
                    :name unused,
                    :defined-by clojure.core/defn-,
                    :filename "sub1/sub2/dummy.clj",
                    :col 1,
                    :name-col 8,
                    :end-col 18,
                    :row 1}],
 :var-usages [{:end-row 1,
               :name-end-col 7,
               :name-end-row 1,
               :name-row 1,
               :name defn-,
               :filename "sub1/sub2/dummy.clj",
               :from user,
               :macro true,
               :col 1,
               :name-col 2,
               :end-col 18,
               :arity 2,
               :varargs-min-arity 2,
               :row 1,
               :to clojure.core}]}

> clj-kondo --lint sub1 --config "{:output {:format :edn :analysis true}}" | jet --query :analysis --pretty
{:namespace-definitions [],
 :namespace-usages [],
 :var-definitions [{:fixed-arities #{0},
                    :end-row 1,
                    :name-end-col 14,
                    :name-end-row 1,
                    :private true,
                    :name-row 1,
                    :ns user,
                    :name unused,
                    :defined-by clojure.core/defn-,
                    :filename "sub1\\sub2\\dummy.clj",
                    :col 1,
                    :name-col 8,
                    :end-col 18,
                    :row 1}],
 :var-usages [{:end-row 1,
               :name-end-col 7,
               :name-end-row 1,
               :name-row 1,
               :name defn-,
               :filename "sub1\\sub2\\dummy.clj",
               :from user,
               :macro true,
               :col 1,
               :name-col 2,
               :end-col 18,
               :arity 2,
               :varargs-min-arity 2,
               :row 1,
               :to clojure.core}]}
```

**config**
None

**expected behavior**
Consistent use of file separator when reporting on Windows.
Either always Unix style or always Windows style.
